packages/datadog-plugin-sharedb/src/index.js

```diff
@@ -0,0 +1,165 @@
+'use strict'
+
+const MessagesAwaitingResponse = new WeakMap()
+const READABLE_ACTION_NAMES = {
+  hs: 'handshake',
+  qf: 'query-fetch',
+  qs: 'query-subscribe',
+  qu: 'query-unsubscribe',
+  bf: 'bulk-fetch',
+  bs: 'bulk-subscribe',
+  bu: 'bulk-unsubscribe',
+  f: 'fetch',
+  s: 'subscribe',
+  u: 'unsubscribe',
+  op: 'op',
+  nf: 'snapshot-fetch',
+  nt: 'snapshot-fetch-by-ts',
+  p: 'presence-broadcast',
+  pr: 'presence-request',
+  ps: 'presence-subscribe',
+  pu: 'presence-unsubscribe'
+}
+
+function getReadableActionName (action) {
+  const actionName = READABLE_ACTION_NAMES[action]
+  if (actionName === undefined) {
+    return action
+  }
+  return actionName
+}
+
+function getReadableResourceName (readableActionName, collection, query) {
+  if (collection) {
+    readableActionName += ' ' + collection
+  }
+  if (query) {
+    readableActionName += ' ' + JSON.stringify(sanitize(query))
+  }
+  return readableActionName
+}
+
+function sanitize (input) {
+  const output = {}
+
+  if (!isObject(input) || Buffer.isBuffer(input)) return '?'
+
+  for (const key in input) {
+    if (typeof input[key] === 'function') continue
+
+    output[key] = sanitize(input[key])
+  }
+
+  return output
+}
+
+function isObject (val) {
+  return typeof val === 'object' && val !== null && !(val instanceof Array)
+}
+
+/**
+ * @description Handle a "receive" event. This will be a message to the server, from the client or the same server
+ * process.
+ * @param {Tracer} tracer DataDog Tracer.
+ * @param {sharedb} config sharedb plugin configuration.
+ * @param {string} action ShareDB Message action.
+ * @param {Object} agent ShareDB Agent
+ * @param {Object} triggerContext ShareDB trigger context (internal middleware concept).
+ * @param {Function} callback Callback to continue middleware execution.
+ * @param {Function} triggerFn Function to start middleware execution.
+ * @returns {*}
+ */
+function handleReceive (tracer, config, action, agent, triggerContext, callback, triggerFn) {
+  if (triggerContext.data && triggerContext.data.a) {
+    const scope = tracer.scope()
+    const childOf = scope.active()
+    // Call the trigger function to continue the middleware chain.
+    return triggerFn.call(this, action, agent, triggerContext, function wrappedCallback (err) {
+      // When the middleware calls back into us, start a trace.
+      const actionName = getReadableActionName(triggerContext.data.a)
+      tracer.trace(
+        'sharedb.request',
+        {
+          childOf,
+          tags: {
+            'service.name': config.service || `${tracer._service}-sharedb`,
+            'span.type': 'sharedb.request',
+            'span.kind': 'client',
+            'resource.method': actionName,
```

Comment:
This should be `sharedb.action` or similar, basically some property in the `sharedb.*` object.

Reply:
It probably makes sense to also add the other metadata from the resource name as `sharedb.*` tags as well.